cirq-google/cirq_google/api/v2/run_context.proto

```diff
@@ -127,15 +127,37 @@ message DeviceParametersDiff {
     // as index into the strs repeated field.
     int32 name = 2;
   }
+
+  // Param value whose type is not among proto field types supported by
+  // ArgValue. In other words, it is the responsibility of the client codes
+  // to enforce the validity and type consistency of this param value.
+  message GenericValue {
+    // Type description of the value representation. Eg. if the following value
+    // bytes field is a JSON string, type_descriptor can be its JSON namespace;
+    // or if the following value field is a protobuf serialization, type_descriptor
+    // can be its protobuf definition URL.
```

Comment:
Such GenericValue seems to have no option for static checking of its fields.  Any errors and bugs would likely get revealed only at runtime due to problem with invalid `type_descriptor` or `value` content that cannot be decoded.

As an alternative would it be possible to extend [program.ArgValue](https://github.com/quantumlib/Cirq/blob/ee4496229b98c3d6d59afc0a4c0480326d091d6e/cirq-google/cirq_google/api/v2/program.proto#L290-L300) with a `RepeatedComplex complex_values` oneof?

This should cover the tested example of readoutCenters of 2 complex values, and we can add more oneof-s to ArgValue as needed (say, `json_value`) later on, WDYT?

Reply:
ArgValue is used by FloatArg for many Gate messages, eg in https://github.com/quantumlib/Cirq/blob/6037f184634605125c11e3690739b6dd18936cad/cirq-google/cirq_google/api/v2/program.proto#L130-L137
the type dependency chain is ```ArgValue --> Arg --> ArgFunction --> FloatArg```

if we add additional oneof's to ArgValue, we have to deal with such value types in the Gate parameters, which may not make sense.

also the ```GenericValue``` type needs to deal with arbitrary key types in Registry sample data, which is very diverse including deep container types such as list of tuples of list of ...... so it would be hard to declare such as proto field types as one-of in ArgValue (incompatible type of Gate parameter issue notwithstanding).

Reply:
ArgValue already has a `string_value` oneof which can be something non-sensible in a Gate context.
I think adding `complex_value`, `complex_values`, and `json_value` oneof-s would not be so much of a stretch; the first two would complete supported number types and the `json_value` string could take in more complicated types encoded as a json.

Would that be sufficient for covering the expected data types?

Reply:
the ```Param.generic_value``` within the oneof is intended to support our internal Registry sample key values, which can be very generic, eg a deeply nested data structure like a list of tuple of value-with-unit.

for internal uses, we intend ```GenericValue.type_descriptor``` to be the labrad type descriptor, eg "(is(vv))" for ```tuple[int, str, tuple[float, float]]```, and ```GenericValue.value``` to be the labrad serialized bytes. This is internal so I refrain from putting such in the fields' docstring.